shared/modules/logger_module.go

```diff
@@ -0,0 +1,19 @@
+package modules
+
+//go:generate mockgen -source=$GOFILE -destination=./mocks/logger_module_mock.go -aux_files=github.com/pokt-network/pocket/shared/modules=module.go
+import (
+	"github.com/rs/zerolog"
+)
+
+type Logger = zerolog.Logger
+
+type LoggerModule interface {
+	Module
+
+	// TODO(#288): Embed `ObservableModule` inside of the `Module` interface so each module has its own context specific logger
+	ObservableModule
```

Comment:
I think that `LoggerModule` _maybe_ is not also an `ObservableModule`.

```suggestion
```

Where would you use `LoggerModule.GetLogger()` if in each module you are getting the `LoggerModule` from the bus and then you are calling `CreateLoggerForModule(someModule.GetModuleName())` in *their* `GetLogger`?

The other modules, yes, they are. And I also like the `Global` logger to log for all the other places. 

If you agree with me, I'd also remove these

```go
func (m *loggerModule) InitLogger() {
	m.logger = m.CreateLoggerForModule(m.GetModuleName())
}

func (m *loggerModule) GetLogger() modules.Logger {
	return m.logger
}
```
but I appreciate the fact that they show how you would use them in a module.

Reply:
Ty @deblasis left a comment on #288 just in case to capture:

> I believe that when the rubber hits the road in https://github.com/pokt-network/pocket/issues/288 there might be (or not) some small changes to this but the foundation looks pretty good to me.